pymc/model_graph.py

```diff
@@ -32,85 +33,64 @@
-                return reversed(node.owner.inputs)
-            else:
-                return None
-
-        list(walk(deque([func]), _expand, bfs=True))
-        return retval
-
-    def _filter_parents(self, var, parents) -> Set[VarName]:
-        """Get direct parents of a var, as strings"""
-        keep = set()  # type: Set[VarName]
-        for p in parents:
-            if p == var:
-                continue
-            elif p.name in self.var_names:
-                keep.add(p.name)
-            elif p in self.transform_map:
-                if self.transform_map[p] != var.name:
-                    keep.add(self.transform_map[p])
-            else:
-                raise AssertionError(f"Do not know what to do with {get_var_name(p)}")
-        return keep
-
-    def get_parents(self, var: TensorVariable) -> Set[VarName]:
-        """Get the named nodes that are direct inputs to the var"""
-        # TODO: Update these lines, variables no longer have a `logpt` attribute
-        if hasattr(var, "transformed"):
-            func = var.transformed.logpt
-        elif hasattr(var, "logpt"):
-            func = var.logpt
-        else:
-            func = var
 
-        parents = self._get_ancestors(var, func)
-        return self._filter_parents(var, parents)
+    def get_parent_names(self, var: TensorVariable) -> Set[VarName]:
+        if var.owner is None or var.owner.inputs is None:
+            return set()
+
+        def _expand(x):
+            if x.name:
+                return [x]
+            if isinstance(x.owner, Apply):
+                return reversed(x.owner.inputs)
+            return []
+
+        parents = {get_var_name(x) for x in walk(nodes=var.owner.inputs, expand=_expand) if x.name}
+
+        return parents
+
+    def vars_to_plot(self, var_names: Optional[Iterable[VarName]] = None) -> List[VarName]:
+        if var_names is None:
+            return self._all_var_names
+
+        selected_names = set(var_names)
+
+        # .copy() because sets cannot change in size during iteration
+        for var_name in selected_names.copy():
+            if var_name not in self._all_var_names:
+                raise ValueError(f"{var_name} is not in this model.")
+
+            for model_var in self.var_list:
+                if hasattr(model_var.tag, "observations"):
+                    if model_var.tag.observations == self.model[var_name]:
+                        selected_names.add(model_var.name)
 
-    def make_compute_graph(self) -> Dict[str, Set[VarName]]:
+        selected_ancestors = set(
+            filter(
+                lambda rv: rv.name in self._all_var_names,
+                list(ancestors([self.model[var_name] for var_name in selected_names])),
+            )
+        )
+
+        for var in selected_ancestors.copy():
+            if hasattr(var.tag, "observations"):
+                selected_ancestors.add(var.tag.observations)
+
+        # ordering of self._all_var_names is important
```

Comment:
I know this has been merged, but what's with this comment? I don't think this will be deterministically ordered since the input comes from a set.

Reply:
I'm not sure anymore... it's been a while since I wrote this. I guess we can remove it